cocotbext/axi/axis.py

```diff
@@ -306,6 +323,14 @@ def __init__(self, bus, clock, reset=None, reset_active_level=True,
             self.byte_lanes = len(self.bus.tkeep)
             if byte_size is not None or byte_lanes is not None:
                 raise ValueError("Cannot specify byte_size or byte_lanes if tkeep is connected")
+            if hasattr(self.bus, "tstrb") and (len(self.bus.tstrb) != self.byte_lanes):
+                raise ValueError("tstrb must be the same width as tkeep")
+            
+        elif hasattr(self.bus, "tstrb"):
```

Comment:
I tstrb only provided, we can use that to get byte_lanes. Also reflected in the docs